src/client/dfs/dfs.c

```diff
@@ -2147,16 +2147,69 @@ dfs_disconnect(dfs_t *dfs)
 	return rc;
 }
 
+static int
+get_uid_gid_from_ownership(daos_handle_t coh, dfs_t *dfs)
+{
+	daos_prop_t		*owner_props;
+	struct daos_prop_entry	*entry;
+	int			rc = 0;
+
+	owner_props = daos_prop_alloc(2);
+	if (owner_props == NULL)
+		return ENOMEM;
+
+	owner_props->dpp_entries[0].dpe_type = DAOS_PROP_CO_OWNER;
+	owner_props->dpp_entries[1].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+
+	rc = daos_cont_query(coh, NULL, owner_props, NULL);
+	if (rc) {
+		D_ERROR("daos_cont_query() failed, "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc = daos_der2errno(rc));
+	}
+
+	/* Convert the owner information to uid/gid */
+	entry = daos_prop_entry_get(owner_props, DAOS_PROP_CO_OWNER);
+	D_ASSERT(entry != NULL);
+	rc = daos_acl_principal_to_uid(entry->dpe_str, &dfs->uid);
+	if (rc == -DER_NONEXIST)
+		/** Set uid to nobody */
+		rc = daos_acl_principal_to_uid("nobody@", &dfs->uid);
+	if (rc) {
+		D_ERROR("Unable to convert owner to uid "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc = daos_der2errno(rc));
+	}
+
+	entry = daos_prop_entry_get(owner_props, DAOS_PROP_CO_OWNER_GROUP);
+	D_ASSERT(entry != NULL);
+	rc = daos_acl_principal_to_gid(entry->dpe_str, &dfs->gid);
+	if (rc == -DER_NONEXIST)
+		/** Set gid to nobody */
+		rc = daos_acl_principal_to_gid("nobody@", &dfs->gid);
+	if (rc) {
+		D_ERROR("Unable to convert owner to gid "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc = daos_der2errno(rc));
+	}
+
+out:
+	daos_prop_free(owner_props);
+	return rc;
+}
+
 int
 dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 {
-	dfs_t			*dfs;
-	daos_prop_t		*prop;
-	struct daos_prop_entry	*entry;
-	struct daos_prop_co_roots *roots;
-	struct dfs_entry	root_dir;
-	int			amode;
-	int			rc;
+	dfs_t				*dfs;
+	daos_prop_t			*prop;
+	struct daos_prop_entry		*entry;
+	struct daos_prop_co_roots	*roots;
+	struct dfs_entry		root_dir;
+	int				amode;
+	int				rc;
+	int				i;
+	#define				NUM_PROPS 3
```

Comment:
this sounds weird to have a #define in here. maybe just use 3 without a define ..
or better just add an int that is 3 or 5 depending on the layout?

Reply:
I was trying to have something that would flag at compile time if the array was the wrong length. I agree it looks a bit weird to have a #define here though.

Reply:
i think there is no need to have a variable. it's just use right below in 2 places, so just using 3 is fine IMO